### PR TITLE
Simplify FCS SurfaceAreaTest test 

### PR DESCRIPTION
**Pull request by smoothdeveloper**
by generating the actual file next to the expected one, which can then be diffed.

I just moved the contents, there is no change in the surface area. Hopefully green.

### CODE REVIEW

File: tests/FSharp.Compiler.Service.Tests/LibraryTestFx.fs
Comment (vzarytovskii) [inline tests/FSharp.Compiler.Service.Tests/LibraryTestFx.fs:70]: All filesystem operations should be done via the FileSystem, could you please change it? There should be some extensions to stream to read/write all.

**Review by vzarytovskii — CHANGES_REQUESTED** [review]

File: tests/FSharp.Compiler.Service.Tests/LibraryTestFx.fs
Comment (smoothdeveloper) [reply tests/FSharp.Compiler.Service.Tests/LibraryTestFx.fs:70]: @vzarytovskii understood, sorry as I wanted to move the file to diff separately from the test.

I'll look at fixing what you point, using the component you describe.

Otherwise, is there any issue with what I suggest to do? 

It makes it easier to diff/merge, run test again, and track the changes to have a separate file.

**Review by vzarytovskii — CHANGES_REQUESTED** [review]
We should not be extending IfileSystem with actual file read/write functionality.
There is StreamExtensions type in the same file which will allow you to do the same (FileSystem.OpenFileForWrite(path).Write(contents)).

**Comment by smoothdeveloper** [comment]
@vzarytovskii makes sense and done.

There is only one downside I see with raw `Stream` extensions versus having an abstraction that would remove the concern of 'use/IDisposable' being handled properly.

This may be improved by having extensions to IFileSystem that would deal with it, taking a `Stream -> 't` function instead of returning the raw `Stream`.

Since this is going to be used in all tooling, turning the API this way (for some use cases at least) would reduce concern for tooling on top of FCS to misuse the resources the concrete implementation returns.

**Comment by smoothdeveloper** [comment]
Green.

**Comment by Happypig375** [comment]
Bump

**Comment by smoothdeveloper** [comment]
Hey @Happypig375, not sure this is a priority for being merged.

I like the change because it skips recompiles but it is not a big deal and probably not urgent for the maintainers to adjust it (also, conficting files now, so can't be merged as is).

I had the pending PR with init properties which is more important that I intended to update based off this, but I can also unblock it by manually adjusting the surface area .fs file there.

**Review by KevinRansom — APPROVED** [review]

**Comment by smoothdeveloper** [comment]
@vzarytovskii thanks! it looks green & good to me.

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**
